### ISSUE TITLE
SC2094 triggers when writing to file early in pipe and reading lazily later in the pipe

### ISSUE THREAD

**Issue by drevicko** [koalaman/shellcheck#2184]
#### For bugs
- Rule Id SC2094
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
python foo.py 2> "$1".txt | less -S - "$1".txt
```

#### Here's what shellcheck currently says:
```
Line 2	SC2094: Make sure not to read and write the same file in the same pipeline.
Line 2	SC2094: Make sure not to read and write the same file in the same pipeline.
```
#### Here's what I wanted or expected to see:

I'd expect there to be no problems reported. 
This is fine because less is lazy when reading `"$1".txt`.

With a target file reader that is not lazy (in the sense of not expecting the whole file to be present immediately), I could envisage a race condition however.

**Comment by TinCanTech** [comment]
The warning seems to be accurate though ..

**Comment by drevicko** [comment]
Perhaps.. Maybe if we add an example where the race condition will be problematic when writing a file early in the pipe and reading later (all the examples are the other way around, and truncation is the issue and easy to make happen). 

I put this issue up because in my use case (getting less to look at stdout and stderr as separate views), there really isn't a problem, and the description confused me some.

Is it worth suggesting people use a named pipe?

**Comment by TinCanTech** [comment]
Or you could ignore it ..

**Comment by drevicko** [comment]
Fair enough.

If I find a moment, I might suggest a small addition to the warning.

**Comment by TinCanTech** [comment]
You can edit the wiki: https://github.com/koalaman/shellcheck/wiki/SC2094 :cake:
